app/src/main/java/io/homeassistant/companion/android/notifications/MessagingManager.kt

```diff
@@ -1745,6 +1764,37 @@ class MessagingManager @Inject constructor(
         }
     }
 
+    private fun togglePersistentConnection(mode: String) {
+        val settingsDao = AppDatabase.getInstance(context).settingsDao()
+
+        when (mode.uppercase()) {
+            WebsocketSetting.NEVER.name -> {
+                settingsDao.get(0)?.let {
+                    it.websocketSetting = WebsocketSetting.NEVER
+                    settingsDao.update(it)
+                }
+            }
+            WebsocketSetting.ALWAYS.name -> {
+                settingsDao.get(0)?.let {
+                    it.websocketSetting = WebsocketSetting.ALWAYS
+                    settingsDao.update(it)
+                }
+            }
+            WebsocketSetting.HOME_WIFI.name -> {
+                settingsDao.get(0)?.let {
+                    it.websocketSetting = WebsocketSetting.HOME_WIFI
+                    settingsDao.update(it)
+                }
+            }
+            WebsocketSetting.SCREEN_ON.name -> {
+                settingsDao.get(0)?.let {
+                    it.websocketSetting = WebsocketSetting.SCREEN_ON
+                    settingsDao.update(it)
+                }
+            }
+        }
```

Comment:
We should also call `WebsocketManager.start(context)` so the service can stop/start immediately, just like we do on the settings page